### PR TITLE
Javascript refactoring

### PR DESCRIPTION
**Pull request by stimms**
Doing a bunch of work with Javascript. Extracting a lot of the edit bug JavaScript and pulling out some components.

### CODE REVIEW

File: Blog x - JavaScript.md [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+
+```
+this.new_posts = document.querySelector('td[name=new_post]')
+```
+
+which makes use of the selector API. jQuery will defer to this behavior if it is available and not buggy in the browser.
+
+###Breaking up JavaScript
+
+EditBug.js is over 400 lines long. While I don't have a particular limit in mind before a class becomes unmaintainable the nature of the code in this file gives me pause. The function are small but they are also largely unrelated. The past majority of the functions don't call into other function which makes me think that the internal coupling in the class is low and so the class is not cohesive. Cohesion is important as it makes replacing the functionality in the future much easier. We can be confident that changes to a class that does nothing but handle e-mail address verification is not likely to change how web URLs are routed.
```
Comment (dpaquette) [inline Blog x - JavaScript.md]: Do you mean the _vast_ majority?

File: Blog x - JavaScript.md [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+```
+$(".warn").click(function (event) {
+                return _this.warn_if_dirty(event);
+            });
+```
+
+Much of the logic related to dirty flags has now be moved out of the EditBug.js file and into a control that can be reused whenever we wish.
+
+
+[View the commit - Adding dirty flag](https://github.com/dpaquette/BugTracker.NET/commit/931220e3fab3c45430e987ea6ee8f066b3edd2f8)
```
Comment (dpaquette) [inline Blog x - JavaScript.md]: This link appears to be broken

File: Blog x - JavaScript.md [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+<%: System.Web.Optimization.Scripts.Render("~/bundles/all") %>
+```
+
+Now when we got to a page we can see that the scripts are indeed all loaded
+
+![Individual scripts](Images/javascript in debug.jpg)
+
+In development all the scripts are included individually to aide in debugging. In production, which is achieved by toggling the flag in the web.config, the scripts are bundled
+
+![Bundled scripts](Images/javascript in release.jpg)
```
Comment (dpaquette) [inline Blog x - JavaScript.md]: I'm not seeing the bundled file in this image

**Comment by dpaquette** [comment]
Excellent post! I had a couple notes on the blog post file itself. Should be ready to merge after you take a look at those.

**Comment by stimms** [comment]
I've addressed those comments.

**Pull request merged**